prov/efa/src/efa_mr.c

```diff
@@ -60,39 +63,39 @@ static struct fi_ops efa_mr_cache_ops = {
 int efa_mr_cache_entry_reg(struct ofi_mr_cache *cache,
 			   struct ofi_mr_entry *entry)
 {
-	int fi_ibv_access = IBV_ACCESS_LOCAL_WRITE;
-
-	struct efa_mr *md = (struct efa_mr *)entry->data;
-
-	md->domain = container_of(cache->domain, struct efa_domain,
-				  util_domain);
-	md->mr_fid.fid.ops = &efa_mr_cache_ops;
-
-	md->mr_fid.fid.fclass = FI_CLASS_MR;
-	md->mr_fid.fid.context = NULL;
-
-	md->mr = ibv_reg_mr(md->domain->ibv_pd, entry->info.iov.iov_base,
-			    entry->info.iov.iov_len, fi_ibv_access);
-	if (!md->mr) {
-		EFA_WARN_ERRNO(FI_LOG_MR, "ibv_reg_mr", errno);
-		return -errno;
-	}
-
-	md->mr_fid.mem_desc = (void *)(uintptr_t)md->mr->lkey;
-	md->mr_fid.key = md->mr->rkey;
-
-	return 0;
+	int ret = 0;
+	/* TODO
+	 * Since, access is not passed as a parameter to efa_mr_cache_entry_reg,
+	 * for now we will set access to all supported access modes i.e.
+	 * FI_SEND | FI_RECV | FI_REMOTE_READ | FI_REMOTE_WRITE. Once access
+	 * information is available this can be removed.
+	 */
+	uint64_t access = FI_SEND | FI_RECV | FI_REMOTE_READ | FI_REMOTE_WRITE;
+
+	struct efa_mr *efa_mr = (struct efa_mr *)entry->data;
+
+	efa_mr->domain = container_of(cache->domain, struct efa_domain,
+				      util_domain);
+	efa_mr->mr_fid.fid.ops = &efa_mr_cache_ops;
+
+	efa_mr->mr_fid.fid.fclass = FI_CLASS_MR;
+	efa_mr->mr_fid.fid.context = NULL;
+
+	ret = efa_mr_reg_impl(entry->info.iov.iov_base,
+				entry->info.iov.iov_len,
+				access, 0, 0, efa_mr, NULL);
+	return ret;
```

Comment:
Would you please run checkpatch?

Reply:
I did run checkpatch, it doesnt give me any errors. I removed it manually. Also rebased to master